main.js

```diff
@@ -0,0 +1,54 @@
+// DOM utility functions
+const addClass = (el, newClass) => {
+  el.className += ` ${newClass}`;
+};
+
+const removeClass = (el, oldClass) => {
+  const currentClasses = el.className.split(" ");
+
+  el.className = currentClasses.filter(names =>
+    names.trim() !== "" && names !== oldClass
+  ).join(" ");
+};
+
+// Search Script
+(() => {
+  const inputField = document.getElementsByTagName('input')[0];
+  let typeTimeout;
+  const TYPE_INTERVAL = 10000;
+
+  inputField.addEventListener("input", (event) => {
+    const value = event.target.value;
+    if (typeTimeout) window.clearTimeout(typeTimeout);
+
+    if (value) {
+      typeTimeout = delay(matchKeyword(value), TYPE_INTERVAL);
+    }
+  }, false);
+
+  const matchKeyword = (value) => {
+    let matchedEmojis;
+    const matchedObjects = EMOJIS.filter(filterByKeyword.bind(this, value));
```

Comment:
Yeah that's a really good point. I'll add comments after in a separate PR so we can discuss separately.